modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -1081,12 +1120,19 @@ public void visit(IfElseStmt ifElseStmt) {
 
             Statement elseIfBody = elseIfBlock.getElseIfBody();
             elseIfBody.accept(this);
+
+            stmtReturns &= elseIfBody.resultsInAReturn();
         }
 
         Statement elseBody = ifElseStmt.getElseBody();
         if (elseBody != null) {
             elseBody.accept(this);
+            stmtReturns &= elseBody.resultsInAReturn();
+        } else {
+            stmtReturns = false;
```

Comment:
Because we can definitively say that a if-else statement causes a function to return only when there are return statements inside all if, if else and else blocks. If at least one such block does not contain the return statement, the whole if-else statement is marked as a statement which will not cause a function to return. In this case, absence of an else block => absence of a return statement.